File: backend/app/models/photo.py
import uuid
from typing import TYPE_CHECKING

from sqlalchemy import ForeignKey, Integer, String
from sqlalchemy.dialects.postgresql import UUID
from sqlalchemy.orm import Mapped, mapped_column, relationship

from app.db.base import Base, UUIDPrimaryKey

if TYPE_CHECKING:
    from app.models.event import Event


class Photo(UUIDPrimaryKey, Base):
    __tablename__ = "photos"

    event_id: Mapped[uuid.UUID] = mapped_column(
        UUID(as_uuid=True),
        ForeignKey("events.id", ondelete="CASCADE"),
        nullable=False,
        index=True,
    )
    image_url: Mapped[str] = mapped_column(String(500), nullable=False)
    thumb_url: Mapped[str | None] = mapped_column(String(500))
    caption: Mapped[str | None] = mapped_column(String(300))
    alt_text: Mapped[str | None] = mapped_column(String(300))
    sort_order: Mapped[int] = mapped_column(Integer, nullable=False, default=0, server_default="0")

    event: Mapped["Event"] = relationship(back_populates="photos")

File: client_app/lib/models/photo.dart
class EventPhoto {
  EventPhoto({
    required this.id,
    required this.eventId,
    required this.imageUrl,
    required this.thumbUrl,
    required this.caption,
    required this.altText,
    required this.sortOrder,
  });

  final String id;
  final String eventId;
  final String imageUrl;
  final String? thumbUrl;
  final String? caption;
  final String? altText;
  final int sortOrder;

  factory EventPhoto.fromJson(Map<String, dynamic> json) {
    return EventPhoto(
      id: json['id'] as String,
      eventId: json['event_id'] as String,
      imageUrl: json['image_url'] as String,
      thumbUrl: json['thumb_url'] as String?,
      caption: json['caption'] as String?,
      altText: json['alt_text'] as String?,
      sortOrder: json['sort_order'] as int,
    );
  }
}
